### DISCUSSION TITLE
Z prob touch

### DISCUSSION THREAD

**Discussion by ThibaudetTevai** [fra589/grbl-Mega-5X#187] [Q&A] [answered]
Salut tous ! 

J'ai vue dans une discussion close que le Z probe de marcher pas . La discution date de 2019. 

Quand est t'il aujourd'hui ?

Merci :)
@fra589 

-----------------

Hi every one 

I found a discussion about the Z touching probe is not inclued in thi repository. What about now ? it's work ?

Thanks :)

**Comment by fra589** [comment, marked as answer]
Bonjour @ThibaudetTevai,  
  
Le palpage est entièrement fonctionnel dans grbl-Mega-5X avec l'ensemble des fonctions G38.2, G38.3, G38.4 et G38.5.  
J'ai récemment ajouté ces fonctionnalités à mon logiciel de contrôle / expéditeur gcode cn5X ++, tu peux voir le résultat ici:  
https://www.facebook.com/groups/238828216699975/permalink/837324230183701/  
  
Probing is fully functional in grbl-Mega-5X with all G38.2, G38.3, G38.4 and G38.5 functions.  
I recently added those functionalities to my cn5X++ control panel/gcode sender, you can see the result here: 
https://www.facebook.com/groups/238828216699975/permalink/837324230183701/  
  
@++;  
Gauthier.

**Reply by ThibaudetTevai** [reply]
Merci pour ta réponse :) 

J'utilise CNCjs actuellement pour faire le palpage. Mon problème est qu'il ne détecte pas le toucher.

Je suis en define :

- CPU_MAP_2560_RAMPS_BOARD

Et j'ai essayer avec et sans ces defines . Mais rien est detecté
-  #define DISABLE_HW_LIMITS_INTERRUPT
-  #define ENABLE_RAMPS_HW_LIMITS

J'ai oublié peut être quelque chose ?

**Reply by ThibaudetTevai** [reply]
Mea Culpa , j'ai confondu la pin A15 avec D15 dans les commentaires du code :)

**Reply by fra589** [reply]
Bonjour @ThibaudetTevai,  
  
Il y à plus simple que la lecture du code pour connaitre le brochage par défaut... J'ai écrit la page de wiki ad hoc pour cela :  
https://github.com/fra589/grbl-Mega-5X/wiki/grbl-Mega-5X-pinout

Ça sert à quoi qu'on se décarcasse ?! 😄 

P.S. Pour utiliser le palpage, il n'y a pas besoin d'activer `#define ENABLE_RAMPS_HW_LIMITS` ni de modifier la valeur par défaut de `#define DISABLE_HW_LIMITS_INTERRUPT`

@++;  
Gauthier.
